frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx

```diff
@@ -39,36 +45,34 @@ export const meta: MetaFunction<typeof loader> = mergeMeta(({ data }) => {
   return data ? getTitleMetaTags(data.meta.title) : [];
 });
 
-export async function loader({ context: { session }, request }: LoaderFunctionArgs) {
+export async function loader({ context: { session }, request, params }: LoaderFunctionArgs) {
   featureEnabled('edit-personal-info');
+
   const instrumentationService = getInstrumentationService();
   const raoidcService = await getRaoidcService();
-  const userService = getUserService();
 
   await raoidcService.handleSessionValidation(request, session);
 
-  const userId = await userService.getUserId();
-  const userInfo = await userService.getUserInfo(userId);
-
-  if (!userInfo) {
-    instrumentationService.countHttpStatus('phone-number.edit', 404);
-    throw new Response(null, { status: 404 });
-  }
-
   const csrfToken = String(session.get('csrfToken'));
+  const personalInformationRouteHelper = getPersonalInformationRouteHelpers();
 
+  const userInfoToken: UserinfoToken = session.get('userInfoToken');
+  const personalInformation = await personalInformationRouteHelper.getPersonalInformation(userInfoToken, params, request, session);
+
+  session.set('personalInformation', personalInformation);
   const t = await getFixedT(request, handle.i18nNamespaces);
   const meta = { title: t('gcweb:meta.title.template', { title: t('personal-information:phone-number.edit.page-title') }) };
 
   instrumentationService.countHttpStatus('phone-number.edit', 200);
-  return json({ csrfToken, meta, userInfo });
+  return json({ csrfToken, meta, personalInformation });
 }
 
 export async function action({ context: { session }, params, request }: ActionFunctionArgs) {
   const log = getLogger('phone-number/edit');
-
+  const personalInformationService = await getPersonalInformationService();
   const instrumentationService = getInstrumentationService();
   const raoidcService = await getRaoidcService();
+  const personalInformation: PersonalInfo = session.get('personalInformation');
```

Comment:
We should be calling the `personalInformationRouteHelper.getPersonalInformation(..)` function to get the `personalInformation` instead.

Reply:
done